lib/PuppeteerSharp.Tests/PuppeteerBaseTest.cs

```diff
@@ -36,5 +36,24 @@ public PuppeteerBaseTest()
         }
 
         public void Dispose() => DisposeAsync().GetAwaiter().GetResult();
+
+        protected static Task<dynamic> WaitForEvents(Session emitter, string eventName, int eventCount = 1)
+        {
+            var completion = new TaskCompletionSource<dynamic>();
```

Comment:
Cool. you attached to the generic Message. I was wondering how to implement this when I implemented the WaitForError. Good Job